### ISSUE TITLE
Internet Status shows "not connected" even when connected + can't update because of this

### ISSUE THREAD

**Issue by Hew-ux** [batocera-linux/batocera-emulationstation#1102]
In **MAIN MENU** -> **NETWORK SETTINGS**, the **INTERNET STATUS** always says "NOT CONNECTED", even if there is an IP address available and showing. The internet connection itself is fine, I'm able to download themes, scrape for media, etc. successfully.

I theorise that because of the internet status claiming to be not connected, that the menu tool for upgrading also fails. So, boot up Batocera, it has a notification that an update is available (I assume that it checks the internet for that) and attempting to download the update results in "NOT CONNECTED", failing the update. It is still possible to update using `batocera-upgrade` via the command line, so this issue is purely with ES's method of upgrading.

Here's the result of `ifconfig`:
![Capture](https://user-images.githubusercontent.com/67527064/148334959-aa299552-4415-4949-ba8d-77386a83e4d6.PNG)

This happens no matter if I'm using a wireless connection, an ethernet cable, both at once, only one at a time, on PC/RPi3/RPi4/s905x3. It's all the exact same issue.

**Comment by fabricecaruso** [comment]
Probably not directly related to ES. This piece of code in ES is very old now. It's using getifaddrs. Something must have changed at system level. The only way to know is to modify code & place logs in queryIPAdress() located in es-core/platform.cpp.
Can you do such a thing or not ?

**Comment by Hew-ux** [comment]
Just tried making my own build of ES to test such changes:
![screenshot-2022 03 21-17h04 32](https://user-images.githubusercontent.com/67527064/159212165-a08373c4-e171-456b-9c19-5177f441cbee.png)
Now here's the funny thing, I have not made any modifications. Was there a PR made to fix this which didn't tag this issue, or was this fixed merely by me compiling my own version of ES?

**Comment by Hew-ux** [comment]
Yep, went back to the regular beta and it seems to be functional now. Therefore, this issue shall be closed!

**Comment by Frontalmule** [comment]
What did you do in order to solve your problem. It seems like it solved it self. I am having what appears to be the same issue. I have tried on RPi4 and N2+. With older versions and using the Beta version and I cannot get Connected.

**Comment by Hew-ux** [comment]
@Frontalmule  I'm sorry to say but it really was like that. Considering that ES uses an old method of detecting an internet connection, it seems to be really inconsistent. Maybe it will one day start working for you too.

At the least, you could use the command line to update.

**Comment by Hew-ux** [comment]
I have found the causation of the issue. On my router, I had previous set a network-wide block on Google's DNS addresses. I had noticed that my configuration had reset this at some point, so I went and turned it back on. Then I immediately noticed I could no longer update Batocera, "Internet connection required". Here are the IPs that are blocked:

**Comment by fabricecaruso** [comment]
We internally use "timeout 1 ping -c 1 -t 1000 8.8.8.8" if we're connected...
So... just don't block google DNS....

**Comment by Hew-ux** [comment]
Eh? Wouldn't it make more sense to ping batocera.org? Especially considering that you aren't downloading the updates from Google, but Batocera. Why build in an additional dependency when it's not even required?

It's also unreliable to just depend on one server, as is the case here. There's a reason many DNS servers have backups.

On that note, pinging a DNS server to detect online connectivity?

**Comment by nadenislamarre** [comment]
emulationstation is not used only by batocera.
getting a resource from internet is the best way to check it is here i think.
but why to block google dns ? i agree it is a good resource to test, even not perfect, even if a new depdendancy, but in any case we need a server dependancy that is very stable.
can you explain why you block google dns ?

**Comment by Hew-ux** [comment]
I'm aware. But to ask the question in the same vein, why Google's DNS server? There are some countries that actually block Google's DNS so it's not a good global solution. At least if it's batocera.org you're more likely to not have this random breakage.

I think it's good to test blocking Google's DNS to find bad practices like this.

"getting a resource from internet is the best way to check it is here i think." -> yeah so make it check with the server you're actually going to be downloading stuff from.

**Comment by Hew-ux** [comment]
To elaborate further, pinging a DNS is not a good way to check for general connectivity. There is the situation where a user could have DNS connectivity but not HTTP connectivity, they use different protocols. If you absolutely must ping Google then ping their http address at google.com, not their DNS address.

**Comment by nadenislamarre** [comment]
whatever is done can be discussed, but we will always fall in somebody having a router configured in a way it doesn't work (except if for batocera we just check the batocera.org, but this is true for updates, maybe not for all).
we may eventually add an option to bypass, but i still don't get why you block the google dns.

**Comment by Hew-ux** [comment]
As I said I use it to test for bad behaviours.

The long of it is that one day I was trying to configure my DNS to use one from a specific location, Vloudflare's 1.1.1.1 address. I set the settings correctly in my router, but found that some of my devices were still using my ISP provided DNS. So I tried many, many different things like setting the DNS on the devices themselves, making a static route on the router to guide the traffic, doing port stuff, nothing was changing it. All I had to go off was that some of my devices were respecting my DNS settings as expected, and others weren't. Eventually that lead me to start blocking the DNSes I didn't want them to be using and found that when Google's DNS was blocked, the misbehaving devices started using the correct DNS, albeit with problems. Eventually I found out that it was because some devices, despite allowing you to input one DNS address, will automatically check Google's DNS if there isn't a second address set. Once Google's DNS was blocked, I found the solution of these devices attempting to search for this address and eventually giving up and switching to my preferred one. Once a second address was set, these devices started behaving as expected and no longer checked Google's DNS, so I could remove all the blocks once again.

So now whenever I'm facing strange internet connectivity issues on particular devices in my network which aren't affecting other devices, I test blocking Google's DNS to see if that's the culprit. The last I had it set to was to block, but my router had undergone the setting possibly during an update which lead to my initial confusion about it.
